tests/utils/mod.rs

```diff
@@ -0,0 +1,264 @@
+            .get(&group_config.ciphersuite)
+            .unwrap();
+        // Initialize the group state for the initial member.
+        let mls_group = MlsGroup::new(
+            &group_id.to_be_bytes(),
+            group_config.ciphersuite,
+            initial_key_package_bundle,
+            group_config.config,
+        )
+        .unwrap();
+        let mut proposal_list = Vec::new();
+        let group_aad = b"";
+        initial_group_member
+            .group_states
+            .borrow_mut()
+            .insert(mls_group.context().group_id.clone(), mls_group);
+        // If there is more than one member in the group, prepare proposals and
+        // commit. Then distribute the Welcome and the commit to the other
+        // members.
+        if group_config.members.len() > 1 {
+            let group_states = initial_group_member.group_states.borrow();
+            let mls_group = group_states
+                .get(&GroupId::from_slice(&group_id.to_be_bytes()))
+                .unwrap();
+            for client_id in 1..group_config.members.len() {
+                // Pull a KeyPackage from the key_store for the new member.
+                let next_member_key_package = key_store
+                    .get_mut(&(
+                        group_config.members[client_id].name,
+                        group_config.ciphersuite,
+                    ))
+                    .unwrap()
+                    .pop()
+                    .unwrap();
+                // Have the initial member create an Add proposal using the new
+                // KeyPackage.
+                let add_proposal = mls_group.create_add_proposal(
+                    group_aad,
+                    initial_credential_bundle,
+                    next_member_key_package,
+                );
+                proposal_list.push(add_proposal);
+            }
+            // Create the commit based on the previously compiled list of
+            // proposals.
+            let (commit_mls_plaintext, welcome_option, _key_package_bundle_option) = mls_group
+                .create_commit(
+                    group_aad,
+                    &initial_credential_bundle,
+                    proposal_list.clone(),
+                    false,
+                )
+                .unwrap();
+            let welcome = welcome_option.unwrap();
+            // Distribute the Welcome message to the other members.
+            for client_id in 1..group_config.members.len() {
+                let new_group_member = test_clients
+                    .get(group_config.members[client_id].name)
+                    .unwrap()
+                    .borrow_mut();
+                // Figure out which key package bundle we should use. This is
+                // a bit ugly and inefficient.
+                let member_secret = welcome
+                    .get_secrets_ref()
+                    .iter()
+                    .find(|x| {
+                        match new_group_member
```

Comment:
Sorry, one more iteration: instead of using `.find(y.get_key_package().hash() == x.key_package_hash).is_some()` this is even more succinct:

`.any(|y| y.get_key_package().hash() == x.key_package_hash)`